Update system naming and references (#992)
hashicorp/design-system: website/docs/getting-started/for-designers/partials/best-practices.md

!!! Info

> These best practices have been curated by the Design Systems Team, Figma has it’s own set of [best practices](https://www.figma.com/best-practices/) for using the tool effectively.

Reviewer: ```suggestion
These best practices have been curated by the Design Systems Team, Figma has its own set of [best practices](https://www.figma.com/best-practices/) for using the tool effectively.
```
